### PR TITLE
Add support for routes for the HTTP PATCH method/verb.

### PR DESCRIPTION
**Pull request by nesv**
As asked for in issue #20.

Also included in this pull request is a new testing function to testing HTTP method matches.

### CODE REVIEW

**Comment by yograterol** [comment]
+1 for thist PR

File: mux_test.go [hunk 1]
```diff
@@ -238,3 +238,40 @@ func TestTail(t *testing.T) {
...
+		t.Fatal(err)
+	}
+
+	resp := httptest.NewRecorder()
+	resp.Code = http.StatusMethodNotAllowed
+	p.ServeHTTP(resp, req)
+	if resp.Code != http.StatusMethodNotAllowed {
+		t.Errorf("Expected code 405, got %d", resp.Code)
+	} else {
+		t.Logf("Success! Wanted %d, got %d", http.StatusMethodNotAllowed, resp.Code)
```
Comment (cespare) [inline mux_test.go]: Here and below, delete these `t.Logf`s; I don't think they add anything.

**Comment by cespare** [comment]
LGTM apart from comment.

**Comment by nesv** [comment]
Thanks @cespare! I have made the changes you suggested.

**Comment by cespare** [comment]
@nesv thanks!

Sorry to keep pestering you, but could you also please squash into a single commit and rebase? And then in that commit message add a line "Fixes #20".

**Comment by nesv** [comment]
Not pestering at all, man. The commits will be squashed momentarily.

**Comment by nesv** [comment]
@cespare I messed something up on my end. I'm going to leave this PR closed, and open up a fresh one.

**Pull request closed without merging**